app/lib/app_secrets.rb

```diff
@@ -0,0 +1,19 @@
+class AppSecrets
+  def self.[](key)
+    result = Vault.kv(namespace).read(key)&.data&.fetch(:value) if ApplicationConfig["VAULT_TOKEN"].present?
+    result ||= ApplicationConfig[key]
+
+    result
+  rescue Vault::VaultError
+    ApplicationConfig[key]
+  end
+
+  def self.[]=(key, value)
+    Vault.kv(namespace).write(key, value: value)
+  end
+
+  def self.namespace
+    ENV["VAULT_SECRET_NAMESPACE"]
```

Comment:
this is not defined in the `Envfile`, correct?

Reply:
Nothing is in the Envfile anymore bc I dont what any of them required in production. I issued a quick follow up PR to fix that #8895 when I forgot that Envied requires everything regardless of default in Prod